internal/aws/ec2.go

```diff
@@ -63,7 +64,13 @@ type EC2API interface {
 	DeleteSecurityGroupByID(string) error
 
 	// ClusterSubnets returns the subnets that are tagged for the cluster
-	ClusterSubnets(scheme *string) (util.Subnets, error)
+	ClusterSubnets(scheme string) ([]string, error)
+
+	// ResolveSecurityGroupNames returns the security group ids for a list of security group names and ids
+	ResolveSecurityGroupNames(context.Context, []string) ([]string, error)
```

Comment:
Did the resolveSecurityGroupNames and ResolveSubnets belong to CloudSvc?
My current mental model for cloudSvc is enhancement for aws-sdk-go, which provides better usable API without application specific api.(which should potentially  be used in other controllers as well).
e.g. for ResolveSubnets, it embed ingress controller specific implementation details, like ingress schema, tagged subnets. I'd preferrer declare this method in alb/lb package.

For the GetSecurityGroups, I'd think we can rename to like GetSecurityGroupsByName or GetSecurityGroupsByNames, which returns []*ec2.SecurityGroup or []ec2.SecurityGroup, or map[string]ec2.SecurityGroup, which seems more general